### ISSUE TITLE
parse_log.sh gives wrong output

### ISSUE THREAD

**Issue by yxchng** [BVLC/caffe#5422]
#Iters Seconds TrainingLoss LearningRate
0      60.772177    60.7705s/100  0.0001
100    96.870289    36.0956s/100  0.0001
200    133.719327   36.8478s/100  0.0001
300    170.667324   36.946s/100   0.0001
400    207.608621   36.9394s/100  0.0001
500    244.500362   36.89s/100    0.0001
600    281.548364   37.0463s/100  0.0001
700    318.434129   36.8841s/100  0.0001
800    355.580382   37.1446s/100  0.0001
900    392.528603   36.9466s/100  0.0001
1000   429.473415   36.9432s/100  0.0001
1100   466.495543   37.0206s/100  0.0001
1200   503.244692   36.7477s/100  0.0001
1300   540.270175   37.0241s/100  0.0001
1400   577.266514   36.995s/100   0.0001
1500   614.206808   36.9389s/100  0.0001
1600   651.201857   36.9936s/100  0.0001
1700   688.026594   36.8234s/100  0.0001
1800   725.065849   37.0379s/100  0.0001
1900   761.979795   36.9126s/100  0.0001
2000   859.710876   97.7279s/100  0.0001
2100   896.690779   36.9773s/100  0.0001
2200   933.594652   36.9026s/100  0.0001
2300   970.522537   36.9266s/100  0.0001
2400   1007.543336  37.0196s/100  0.0001
2500   1044.460583  36.916s/100   0.0001
2600   1081.500929  37.0391s/100  0.0001
2700   1118.330184  36.8281s/100  0.0001
2800   1155.352403  37.021s/100   0.0001
2900   1192.409225  37.0556s/100  0.0001
3000   1229.340137  36.9297s/100  0.0001
3100   1266.362365  37.0211s/100  0.0001
3200   1303.306361  36.9428s/100  0.0001
3300   1340.233099  36.9256s/100  0.0001
...

The training loss here is time per 100 iteration.

**Comment by yxchng** [comment]
<img width="990" alt="screen shot 2017-03-18 at 10 51 28 am" src="https://cloud.githubusercontent.com/assets/10518587/24068408/dcf946bc-0bc8-11e7-98ae-f5279062bc4d.png">

I think this is the part that causes the problem. Why is the log format not consistent?

**Comment by yxchng** [comment]
And also why is their two losses for training when I only define one?

1) Iteration ..... , loss = ...
2) Train net output ... : loss = ...

and these 2 losses are of different values.

**Comment by fangbinwei** [comment]
I met this problem,  I changed the parse_log.sh, and it worked
before
`grep ', loss = ' $1 | awk '{print $9}' > aux1.txt
`
after changed
` grep ', loss = ' $1 | awk -F = '{print $2}' > aux1.txt
`
I use "=" to help find the right position of loss, I hope it can help you.

**Comment by shelhamer** [comment]
@cypof this issue and #5361 seem to be caused by 3ba20549b7f49a76cd023d19f781a6891b2c2122 since the solver logging was altered. See https://github.com/BVLC/caffe/blob/master/src/caffe/solver.cpp#L220-L221

Could you fix the tools? (Bonus: add simple tests, like for draw_net.py, or else this will keep breaking accidentally.)

**Comment by cypof** [comment]
thks @fangbinwei your fix works. The python parser looks ok. fix in #5514

@yxchng iteration loss is snoothed, c.f. solfver_param.average_loss.

**Comment by ChrystleMyrnaLobo** [comment]
> And also why is their two losses for training when I only define one?
> 
>     1. Iteration ..... , loss = ...
This value is the average over the last `average_loss`  iterations as specified in the `solver.prototxt` parameter. 
https://github.com/BVLC/caffe/wiki/Solver-Prototxt

> 
>     2. Train net output ... : loss = ...
> 
This value is the loss for that specific iteration
